### PR TITLE
FUSE tests: Submitting at early stage for feedback on how you like 'em

### PR DESCRIPTION
**Pull request by tv42**

### CODE REVIEW

**Comment by tv42** [comment]
There's a little bit of undesired repeating of things on the mount code path, e.g. the tests are not going through the `Mounter` abstraction and aren't necessarily using the same mount options. On the plus side, using the `fstestutil` library makes tests easier to write, and we can work on unifying the code paths later.

File: src/internal/mock_pfs/mock.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
```
Comment (jdoliner) [inline src/internal/mock_pfs/mock.sh]: Should we have a make target for this? Maybe `make mock`?

Comment (tv42) [inline src/internal/mock_pfs/mock.sh]: It's triggered by `go generate`, and that shell script can even go away once the bug referenced below is fixed.

Comment (jdoliner) [inline src/internal/mock_pfs/mock.sh]: I see. Makes sense.

**Comment by tv42** [comment]
Redid the tests to avoid the overly pedantic "captured protocol conversation" style, now all tests run against localhost API instances (managed as part of the test). Please take a look, should be good to merge.

**Comment by jdoliner** [comment]
:+1:

Just ran these tests locally, all is kosher.

Merging.

**Pull request merged**